core/src/rpc.rs

```diff
@@ -425,17 +433,42 @@ impl JsonRpcRequestProcessor {
             .map(|(_, status)| status)
     }
 
-    pub fn get_signature_statuses_with_commitment(
+    pub fn get_signature_statuses(
         &self,
         signatures: Vec<Signature>,
-        commitment: Option<CommitmentConfig>,
+        config: Option<RpcSignatureStatusConfig>,
     ) -> RpcResponse<Vec<Option<TransactionStatus>>> {
         let mut statuses: Vec<Option<TransactionStatus>> = vec![];
 
+        let (commitment, search_transaction_history) = if let Some(config) = config {
+            (
+                config.commitment,
+                config.search_transaction_history.unwrap_or(false),
+            )
+        } else {
+            (None, false)
+        };
         let bank = self.bank(commitment);
 
         for signature in signatures {
-            let status = self.get_transaction_status(signature, &bank);
+            let status = if let Some(status) = self.get_transaction_status(signature, &bank) {
+                Some(status)
+            } else if self.config.enable_rpc_transaction_history && search_transaction_history {
+                self.blockstore
+                    .get_transaction_status(signature)
+                    .map_err(|_| Error::internal_error())?
+                    .map(|(slot, status_meta)| {
+                        let err = status_meta.status.clone().err();
+                        TransactionStatus {
+                            slot,
+                            status: status_meta.status,
+                            confirmations: None,
+                            err,
+                        }
+                    })
+            } else {
```

Comment:
I think if clients request history on a node which has rpc tx history disabled, they should receive an error. But there's no way for a client to query that config information from a node beforehand so maybe we handle this later? I can write an issue

Reply:
Cool, we can return a similar error from `getConfirmedBlock`

Reply:
Issue: https://github.com/solana-labs/solana/issues/9316